### ISSUE TITLE
Because of the work needs, I need to understand PartitionAlloc, this project is very good, can you provide more detailed usage? I don't really understand how to use it.

### ISSUE THREAD

**Issue by eternalsakura** [msmania/chromium-debug#1]
Because of the work needs, I need to understand PartitionAlloc, this project is very good, can you provide more detailed usage? I don't really understand how to use it.
thanks!

**Comment by msmania** [comment]
Have you ever used Windows Debugger?  This is a debugger extension for Windows Debugger.

Getting Started with WinDbg (User-Mode) - Windows drivers | Microsoft Docs
https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/getting-started-with-windbg

I also wrote the following article about PartitionAlloc before.  It might be helpful if Google translate works well.
https://qiita.com/msmania/items/1a10d5b7d20f6455b9b1

**Comment by eternalsakura** [comment]
thanks!I know a little about windbg, but how to install it into windbg?
Do I need to compile this project myself?

**Comment by eternalsakura** [comment]
Can it be used directly to debug release chrome?thanks!

**Comment by msmania** [comment]
Yes, you need to build this project first.  Build steps are easy.  Download the latest Visual Studio Community (or other editions if you can afford to pay for it), open "x86/x64 Native Tools Command Prompt", and run `NMAKE` without any options.  Then you'll get a extension DLL cr.dll, which can be loaded by running the debugger command `.load <path to cr.dll>`.

Loading Debugger Extension DLLs - Windows drivers | Microsoft Docs
https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/loading-debugger-extension-dlls

You don't need to build Chromium on your own.  This works with the official Chrome/ChromeCanary.  To do that, you need to get the private symbol.  Please check the section "Symbol server" in the following page.  Actually if you use your private Chromium, you need to modify the extension's code because the name of Chromium's core module is different and I hardcoded the name as chrome_child.dll.

Debugging Chromium on Windows - The Chromium Projects
https://www.chromium.org/developers/how-tos/debugging-on-windows

The page above introduces some parameters for chrome.exe that might be helpful, but it's just optional.  Just launching Chrome as usual, and attaching your debugger to a render process is good enough.

I have not used this extension for a while.  Probably it doesn't work with the latest Chrome now.  I'll check and update if needed.

**Comment by msmania** [comment]
I've updated PartitionAlloc commands so that you can use it to debug Chrome 70.

**Comment by eternalsakura** [comment]
thanks!
